src/Planners/include/BipedalLocomotion/Planners/SO3Planner.tpp

```diff
@@ -0,0 +1,175 @@
+{
+    if (duration <= 0)
+    {
+        std::cerr << "[SO3Planner::setRotation] The trajectory duration must be a strictly "
+                     "positive number."
+                  << std::endl;
+        return false;
+    }
+
+    m_initialRotation = initialRotation;
+    m_T = duration;
+
+    if constexpr (trivialization == LieGroupTrivialization::Right)
+    {
+        m_distance = (finalRotation * initialRotation.inverse()).log();
+    } else
+    // Please read it as trivialization == LieGroupTrivialization::Left
+    {
+        m_distance = (initialRotation.inverse() * finalRotation).log();
+    }
+
+    // reset the advance current time
+    m_advanceCurrentTime = 0;
+
+    // reset the state
+    m_state.rotation = m_initialRotation;
+    m_state.velocity.setZero();
+    m_state.acceleration.setZero();
+
+    return true;
+}
+
+template <LieGroupTrivialization trivialization>
+template <class Derived>
+bool SO3Planner<trivialization>::evaluatePoint(const double& time,
+                                               manif::SO3d& rotation,
+                                               manif::SO3TangentBase<Derived>& velocity,
+                                               manif::SO3TangentBase<Derived>& acceleration) const
+{
+    if (time < 0 || time > m_T)
+    {
+        std ::cerr << "[SO3Planner::evaluatePoint] The time has to be a real positive number "
+                      "between 0 and "
+                   << m_T << "." << std::endl;
+        return false;
+    }
+
+    const double& t = time;
+    // Compute the displacement in the tangent space
+    // - displacement_tangent = log(finalRotation * initialRotation.inverse()) * s(t) (in case of
+    //                                                                                 Right
+    //                                                                                 Trivialization)
+    // - displacement_tangent = log(initialRotation.inverse() * finalRotation) * s(t) (in case of
+    //                                                                                 Left
+    //                                                                                 Trivialization)
+    // You can find the computation of s in "Modern Robotics: Mechanics, Planning, and Control"
+    // (Chapter 9.2)
+    const double s = 10 * std::pow(t / m_T, 3)
+                     - 15 * std::pow(t / m_T, 4)
+                     + 6 * std::pow(t / m_T, 5);
+    const manif::SO3d::Tangent displacementTangent = m_distance * s;
+
+    if constexpr (trivialization == LieGroupTrivialization::Right)
+    {
+        // please read this as R(t) = exp(displacementTangent) * initialRotation
+        rotation = displacementTangent.exp() * m_initialRotation;
+    } else
+    // Please read it as trivialization == LieGroupTrivialization::Left
+    {
+        // please read this as R(t) = initialRotation * exp(displacementTangent)
+        rotation = m_initialRotation * displacementTangent.exp();
```

Comment:
Looks like a right-plus, to be checked:
```suggestion
        rotation = m_initialRotation + displacementTangent;
```

Reply:
Related comment https://github.com/dic-iit/bipedal-locomotion-framework/pull/84#discussion_r487762652